src/components/views/rooms/UserIdentityWarning.tsx

```diff
@@ -0,0 +1,321 @@
+    // prompt if necessary. The user will only be added if they are actually a
+    // member of the room. If they are not a member, this function will do
+    // nothing.
+    const addMemberNeedingApproval = useCallback(
+        (userId: string, member?: RoomMember): void => {
+            if (userId === cli.getUserId()) {
+                // We always skip our own user, because we can't pin our own identity.
+                return;
+            }
+            member = member ?? room.getMember(userId) ?? undefined;
+            if (member) {
+                membersNeedingApprovalRef.current.set(userId, member);
+                // We only select the prompt if we are done initialising,
+                // because we will select the prompt after we're done
+                // initialising, and we want to start by displaying a warning
+                // for the user with the smallest ID.
+                if (initialisedRef.current === InitialisationStatus.Completed) {
+                    setCurrentPrompt((currentPrompt) => {
+                        // If we aren't currently displaying a warning, we pick
+                        // a new user to show a warning for.  If we are already
+                        // displaying a warning, don't change the display.
+                        //
+                        // We have to do this in a callback to
+                        // `setCurrentPrompt` because this function could have
+                        // been called after an `await`, and the `currentPrompt`
+                        // that this function would have may be outdated.
+                        if (!currentPrompt) {
+                            return selectCurrentPrompt();
+                        } else {
+                            return currentPrompt;
+                        }
+                    });
+                }
+            }
+        },
+        [cli, room, selectCurrentPrompt],
+    );
+
+    // Check if the user's identity needs approval, and if so, add them to the
+    // membersNeedingApproval map and update the prompt if needed. They will
+    // only be added if they are a member of the room.
+    const addMemberIfNeedsApproval = useCallback(
+        async (userId: string, member?: RoomMember): Promise<void> => {
+            const gotVerificationStatusUpdate = gotVerificationStatusUpdateRef.current;
+            const membersNeedingApproval = membersNeedingApprovalRef.current;
+
+            if (gotVerificationStatusUpdate.has(userId)) {
+                // We're already checking their verification status, so we don't
+                // need to do anything here.
+                return;
+            }
+            gotVerificationStatusUpdate.set(userId, false);
+            if (await userNeedsApproval(crypto!, userId)) {
+                if (!membersNeedingApproval.has(userId) && gotVerificationStatusUpdate.get(userId) === false) {
```

Comment:
```suggestion
                // Only actually update the list if we haven't received a `UserTrustStatusChanged` for this user in
                // the meantime.
                if (!membersNeedingApproval.has(userId) && gotVerificationStatusUpdate.get(userId) === false) {
```

Reply:
Also, consider removing `!membersNeedingApproval.has(userId)` from this condition. AFAICT, `addMemberNeedingApproval` is a no-op if the user is already in the list, and removing the check would simplify this function.